x/pylons/keeper/msg_server_execute_recipe_test.go

```diff
@@ -1,82 +1,417 @@
+			valid:                 false,
+		},
+		{
+			decs:                "Invalid PaymentInfo",
+			cookbook:            *cookbook,
+			unEnabledCookbook:   false,
+			changeIdCookbook:    false,
+			recipe:              *recipe,
+			changeIdRecipe:      false,
+			updateCoinInputs:    false,
+			updateEntriesRecipe: false,
+			execution: types.MsgExecuteRecipe{
+				Creator:         types.TestCreator,
+				CookbookId:      "",
+				RecipeId:        "",
+				CoinInputsIndex: 0,
+				ItemIds:         nil,
+				PaymentInfos: []types.PaymentInfo{
+					{
+						PurchaseId:    "1",
+						ProcessorName: "test",
+						PayerAddr:     "pylon0123",
+						Amount:        sdk.OneInt(),
+						ProductId:     "1",
+						Signature:     "test",
+					},
+				},
+			},
+			changeExcutionCreator: false,
+			valid:                 false,
+		},
+		{
+			decs:                "Amount minted reached maximum limit",
+			cookbook:            *cookbook,
+			unEnabledCookbook:   false,
+			changeIdCookbook:    false,
+			recipe:              *recipe,
+			changeIdRecipe:      false,
+			updateCoinInputs:    false,
+			updateEntriesRecipe: true,
+			execution: types.MsgExecuteRecipe{
+				Creator:         types.TestCreator,
+				CookbookId:      "",
+				RecipeId:        "",
+				CoinInputsIndex: 0,
+				ItemIds:         nil,
+			},
+			changeExcutionCreator: false,
+			valid:                 false,
+		},
+		{
+			decs:                "Process PaymentInfo error",
+			cookbook:            *cookbook,
+			unEnabledCookbook:   false,
+			changeIdCookbook:    false,
+			recipe:              *recipe,
+			changeIdRecipe:      false,
+			updateCoinInputs:    false,
+			updateEntriesRecipe: true,
+			execution: types.MsgExecuteRecipe{
+				Creator:         types.TestCreator,
+				CookbookId:      "",
+				RecipeId:        "",
+				CoinInputsIndex: 0,
+				ItemIds:         nil,
+				PaymentInfos: []types.PaymentInfo{{
+					PurchaseId:    "test",
+					ProcessorName: "TestPayment",
+					PayerAddr:     types.GenTestBech32FromString(types.TestCreator),
+					Amount:        sdk.NewInt(2),
+					ProductId:     "testProductId",
+					Signature:     genTestPaymentInfoSignature("testPurchaseId", types.GenTestBech32FromString(types.TestCreator), "testProductId", sdk.NewInt(2), privKey),
+				}},
+			},
+			changeExcutionCreator: false,
+			valid:                 false,
+		},
+		{
+			decs:                "Error lock coins for execution",
+			cookbook:            *cookbook,
+			unEnabledCookbook:   false,
+			changeIdCookbook:    false,
+			recipe:              *recipe,
+			changeIdRecipe:      false,
+			updateCoinInputs:    true,
+			updateEntriesRecipe: false,
+			execution: types.MsgExecuteRecipe{
+				Creator:         types.TestCreator,
+				CookbookId:      "",
+				RecipeId:        "",
+				CoinInputsIndex: 0,
+				ItemIds:         nil,
+			},
+			changeExcutionCreator: false,
+			valid:                 false,
+		},
+		{
+			decs:                "User account username not found",
```

Comment:
"Username not found" can be a better error message.

Reply:
I'm confused as to why usernames are involved at this point, don't we know who everyone is by the signature?

Reply:
It's based on the line of code in `ExecuteRecipe` function @MikeSofaer : https://github.com/Pylons-tech/pylons/blob/1ed56732623153887ab31c4fea11d6708e03a433/x/pylons/keeper/msg_server_execute_recipe.go#L181

Reply:
OK but how can it occur? The creator must have a username, how could we be using the wrong one?

Reply:
As you can see, the case I have still caught that point